### PR TITLE
Issue55222

### LINKED ISSUE #55222: Python 3.7 compatibility re._pattern_type -> re.Pattern
##### SUMMARY

With Python 3.7 we get the following error:
AttributeError: module 're' has no attribute '_pattern_type'

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

lib/ansible/module_utils/network/common/config.py

##### ANSIBLE VERSION

```paste below
ansible 2.7.8
  config file = /home/ops/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible/plugins/modules', '/home/ops/local/ansible-modules']
  ansible python module location = /home/ops/lib/python3.7/site-packages/ansible
  executable location = /home/ops/bin/ansible
  python version = 3.7.0 (default, Jan 16 2019, 12:24:08) [GCC 4.4.7 20120313 (Red Hat 4.4.7-17)]
```

##### CONFIGURATION

```paste below
DEFAULT_ACTION_PLUGIN_PATH(/home/ops/ansible/ansible.cfg) = ['/usr/share/ansible_plugins/action_plugins', '/home/ops/local/ansible-plugins/action']
DEFAULT_ASK_VAULT_PASS(/home/ops/ansible/ansible.cfg) = True
DEFAULT_FILTER_PLUGIN_PATH(/home/ops/ansible/ansible.cfg) = ['/usr/share/ansible_plugins/filter_plugins', '/home/ops/local/ansible-plugins/filter']
DEFAULT_FORKS(/home/ops/ansible/ansible.cfg) = 20
DEFAULT_GATHERING(/home/ops/ansible/ansible.cfg) = explicit
DEFAULT_HOST_LIST(/home/ops/ansible/ansible.cfg) = ['/home/ops/ansible/inventoryyaml']
DEFAULT_INVENTORY_PLUGIN_PATH(/home/ops/ansible/ansible.cfg) = ['/usr/share/ansible_plugins/inventory_plugins', '/home/ops/local/ansible-plugins/inventory']
DEFAULT_LOAD_CALLBACK_PLUGINS(/home/ops/ansible/ansible.cfg) = True
DEFAULT_LOOKUP_PLUGIN_PATH(/home/ops/ansible/ansible.cfg) = ['/usr/share/ansible_plugins/lookup_plugins', '/home/ops/local/ansible-plugins/lookup']
DEFAULT_MODULE_PATH(/home/ops/ansible/ansible.cfg) = ['/usr/share/ansible/plugins/modules', '/home/ops/local/ansible-modules']
DEFAULT_STDOUT_CALLBACK(/home/ops/ansible/ansible.cfg) = yaml
DEFAULT_TRANSPORT(/home/ops/ansible/ansible.cfg) = local
DEFAULT_VARS_PLUGIN_PATH(/home/ops/ansible/ansible.cfg) = ['/usr/share/ansible_plugins/vars_plugins', '/home/ops/local/ansible-plugins/vars']
INVENTORY_ENABLED(/home/ops/ansible/ansible.cfg) = ['yaml']
PERSISTENT_COMMAND_TIMEOUT(/home/ops/ansible/ansible.cfg) = 120
PERSISTENT_CONNECT_RETRY_TIMEOUT(/home/ops/ansible/ansible.cfg) = 60
PERSISTENT_CONNECT_TIMEOUT(/home/ops/ansible/ansible.cfg) = 120
RETRY_FILES_ENABLED(/home/ops/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT

RedHat 6.8

##### STEPS TO REPRODUCE

Running Ansible network_cli with Python 3.7 fails. 

```yaml

```

##### EXPECTED RESULTS

No errors

##### ACTUAL RESULTS

Errors

```paste below
AttributeError: module 're' has no attribute '_pattern_type'
```

### PR DESCRIPTION
**Pull request by ajsiersema**
##### SUMMARY

Python 3.7 compatibility for re.Pattern (Fixes #55222)

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

lib/ansible/module_utils/network/common/config.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/module_utils/network/common/config.py [hunk 1]
```diff
@@ -40,6 +40,10 @@
     re.compile(r"Current configuration : \d+ bytes")
 ])
 
+try:
```
Comment (Qalthos) [inline lib/ansible/module_utils/network/common/config.py:44]:
```suggestion

try:
```

File: lib/ansible/module_utils/network/common/config.py [hunk 2]
```diff
@@ -40,6 +40,10 @@
...
+    Pattern = re._pattern_type
+except AttributeError:
+    Pattern = re.Pattern
```
Comment (Qalthos) [inline lib/ansible/module_utils/network/common/config.py:47]:
```suggestion
    Pattern = re.Pattern

```

**Review by Qalthos — CHANGES_REQUESTED** [review]
Those changes should solve the test issues. Please also remove the unrelated commits from this PR.

File: lib/ansible/module_utils/network/common/config.py (hunk 2 above)
Comment (ajsiersema) [reply lib/ansible/module_utils/network/common/config.py:47]: Sorry about the unrelated commits, not sure how I could have let that happen.

File: lib/ansible/module_utils/network/common/config.py (hunk 1 above)
Comment (ajsiersema) [reply lib/ansible/module_utils/network/common/config.py:44]: Fixed the PR

**Review by Qalthos — APPROVED** [review]

**Pull request merged**